### ISSUE TITLE
support for :not Modifier

### ISSUE THREAD

**Issue by vsocrates** [FHIR/fhir.js#118]
Hi everyone,

I am trying to implement a query using fhir.js that essentially pulls all "Observations" that aren't in a specified set of LOINC codes. I do this with the following URL: 

[https://fhirtest.uhn.ca/baseDstu3/Observation?subject=240225&code:not=30522-7&code:not=38483-4](url)

However, it doesn't seem like this is possible using the fhir.js library as not is not one of the supported Modifiers. I have tried with the following queryObject: 

`{'$and': [{'subject':'240225'},{code: {'$and': [{$not: '30522-7'}, {$not: '20565-8'}, {$not: '6298-4'}]}}]}`

 I would be willing to implement this but I'm not completely sure where it should go. From preliminary searching, I believe I should just need to update the `search.js` middleware file. 

Please let me know if this is the case. I'm sorry, I don't quite have the JS knowledge to know exactly where the changes should go. Thanks!

**Comment by niquola** [comment]
Yes you have to update search middleware! Please provide PR with tests - i will help.
